src/include/random.h

```diff
@@ -0,0 +1,115 @@
+// -*- mode:C++; tab-width:8; c-basic-offset:2; indent-tabs-mode:t -*-
+// vim: ts=8 sw=2 smarttab
+/*
+ * Ceph - scalable distributed file system
+ *
+ * Copyright (C) 2017 SUSE LINUX GmbH
+ *
+ * This is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License version 2.1, as published by the Free Software
+ * Foundation.  See file COPYING.
+ *
+*/
+
+#ifndef CEPH_RANDOM_H
+#define CEPH_RANDOM_H 1
+
+#include <random>
+
+// Basic random number facility, adapted from N3551:
+namespace ceph {
+namespace util {
+
+namespace detail {
+
+template <typename T>
+T& urng()
+{
+ static T rng;
+ return rng;
+}
```

Comment:
note that a static engine is not thread-safe. `Fixes: http://tracker.ceph.com/issues/18873` seems to imply otherwise

to be thread-safe, applications either need to provide mutual exclusion when accessing a shared engine, or guarantee that a given engine is only accessed from a single thread